bpython/test/test_brackets_completion.py

```diff
@@ -0,0 +1,166 @@
+import os
+
+from bpython.test import FixLanguageTestCase as TestCase, TEST_CONFIG
+from bpython.curtsiesfrontend import repl as curtsiesrepl
+from bpython import config
+
+
+def setup_config(conf):
+    config_struct = config.Config(TEST_CONFIG)
+    for key, value in conf.items():
+        if not hasattr(config_struct, key):
+            raise ValueError(f"{key!r} is not a valid config attribute")
+        setattr(config_struct, key, value)
+    return config_struct
+
+
+def create_repl(brackets_enabled=False, **kwargs):
+    config = setup_config(
+        {"editor": "true", "brackets_completion": brackets_enabled}
+    )
+    repl = curtsiesrepl.BaseRepl(config, **kwargs)
+    os.environ["PAGER"] = "true"
+    os.environ.pop("PYTHONSTARTUP", None)
+    repl.width = 50
+    repl.height = 20
+    return repl
+
+
+class TestBracketCompletionEnabled(TestCase):
+    def setUp(self):
+        self.repl = create_repl(brackets_enabled=True)
+
+    def test_start_line(self):
+        self.repl.process_event("(")
+        self.assertEqual(self.repl._current_line, "()")
+        self.assertEqual(self.repl._cursor_offset, 1)
+
+    def test_nested_brackets(self):
+        self.repl.process_event("(")
+        self.repl.process_event("[")
+        self.repl.process_event("{")
+        self.assertEqual(self.repl._current_line, """([{}])""")
+        self.assertEqual(self.repl._cursor_offset, 3)
+
+    def test_quotes(self):
+        self.repl.process_event("(")
```

Comment:
Maybe a helper method here called `self.process_events(["(", "'", "x", "<TAB>"])` would be helpful for readability? Your call, no need to go overboard on making tests DRY.